v3/lints/cabf_smime_br/lint_subscribers_crl_distribution_points_are_http.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * ZLint Copyright 2023 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package cabf_smime_br
+
+import (
+	"net/url"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "e_subscribers_crl_distribution_points_are_http",
+		Description:   "cRLDistributionPoints SHALL have URL scheme HTTP.",
+		Citation:      "7.1.2.3.b",
+		Source:        lint.CABFSMIMEBaselineRequirements,
+		EffectiveDate: util.CABF_SMIME_BRs_1_0_0_Date,
+		Lint:          NewSubscriberCrlDistributionPointsHTTP,
+	})
+}
+
+type subscriberCrlDistributionPointsHTTP struct{}
+
+func NewSubscriberCrlDistributionPointsHTTP() lint.LintInterface {
+	return &subscriberCrlDistributionPointsHTTP{}
+}
+
+func (l *subscriberCrlDistributionPointsHTTP) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && (util.IsMultipurposeSMIMECertificate(c) || util.IsStrictSMIMECertificate(c))
+}
+
+func (l *subscriberCrlDistributionPointsHTTP) Execute(c *x509.Certificate) *lint.LintResult {
+	for _, dp := range c.CRLDistributionPoints {
+		parsed, err := url.Parse(dp)
+		if err != nil {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "SMIME certificate contains invalid CRL distribution point",
+			}
+		}
+		if parsed.Scheme != "http" {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "SMIME certificate contains invalid URL scheme in CRL distribution point",
+			}
+		}
+	}
+	return &lint.LintResult{Status: lint.Pass}
```

Comment:
This looks like it passes if a CRL distribution point is not present.

```
b. cRLDistributionPoints (SHALL be present

This extension SHOULD NOT be marked critical. It SHALL contain at least one distributionPoint...
```
I realize this requirement doesn't perfectly match the lint name, but it is part of 7.1.2.3.b. Is the intention to follow this up with another PR that finishes 7.1.2.3.b?

Reply:
This is covered separately in #742.

Reply:
The rest of 7.1.2.3.b is handled by #742.